### PR TITLE
Adds QUICHE HTTP/2 test targets to CI.

### LINKED ISSUE #21187: quiche: quiche h2 unit tests failure
Some h2 unit tests from QUICHE were added to quiche.BUILD, but they are not tested in CI. And some actually failed if ran manually, i.e. @com_github_google_quiche//:http2_adapter_nghttp2_session_test
repro: $ bazel test @com_github_google_quiche//...

To tell CIs to run those tests, the target should be added to   https://github.com/envoyproxy/envoy/blob/main/bazel/external/quiche.BUILD#L55

### PR DESCRIPTION
**Pull request by birenroy**
Fixes #21187 

Signed-off-by: Biren Roy <birenroy@google.com>

Commit Message: Adds QUICHE HTTP/2 test targets to CI.
Additional Description:
Risk Level: low, test targets only
Testing: Ran affected tests using "bazel test" in both fastbuild and opt mode.
Docs Changes: n/a
Release Notes: n/a, test targets only
Platform Specific Features:

### CODE REVIEW

File: bazel/external/quiche.BUILD [hunk 1]
```diff
@@ -55,8 +55,22 @@ quiche_copts = select({
...
+        "http2_adapter_nghttp2_session_test",
+        "http2_adapter_nghttp2_util_test",
+        "http2_adapter_oghttp2_adapter_test",
+        "http2_adapter_oghttp2_session_test",
+        "http2_adapter_oghttp2_util_test",
+        "http2_adapter_recording_http2_visitor_test",
+        "http2_adapter_window_manager_test",
         "http2_platform_api_test",
         "quiche_common_test",
+        "spdy_core_header_block_test",
```
Comment (danzh2010) [inline bazel/external/quiche.BUILD:73]: Thanks for picking up this! We don't run all QUICHE unit tests in CI due to limited resources.The tests added here are usually platform dependent so not covered by QUICHE continuous testing internally or in chromium, or code added for Envoy specifically. Do these h2 and spdy tests fall into those categories?

Comment (danzh2010) [reply bazel/external/quiche.BUILD:73]: As discussed offline, these tests are light weight enough.

**Review by KBaichoo — APPROVED** [review]
lgtm, thanks

**Pull request merged**